os/kernel/binary_manager/binary_manager_recovery.c

```diff
@@ -207,6 +207,15 @@ void binary_manager_exclude_rtthreads(struct tcb_s *tcb)
 	prev = tcb->bin_blink;
 	next = tcb->bin_flink;
 
+	if (tcb->sched_priority > BM_PRIORITY_MAX) {
```

Comment:
Isn't it duplicated with checking the type of realtime thread at up_assert.c line 460?

Reply:
It should be checked because this tcb can be non real-time task even though binary is real-time.